Sources/Helpers/Factories/CellItemFactory.swift

```diff
@@ -4,36 +4,112 @@
+
+    /// Set this handler to configure the cell item
     ///
-    ///    - object: an object to create a cell item for it
-    func makeCellItem(object: Any) -> CollectionViewCellItem?
+    /// - Parameters:
+    ///    - CellItem: generated universal cell item. Associated object can be retrieved via `cellItem.object`.
+    public var cellItemConfigurationHandler: ((CellItem) -> Void)?
+
+    /// Set this handler to configure the cell
+    ///
+    /// - Parameters:
+    ///    - UICollectionViewCell: the cell that should be configured
+    ///    - CellItem: generated universal cell item. Associated object can be retrieved via `cellItem.object`.
+    public var cellConfigurationHandler: ((Cell, CellItem) -> Void)?
+
+    public init() {
+    }
 
     /// Returns an array of cell items
     ///
     /// - Parameters:
-    ///    - array: an array of objects to create cell items for them
-    func makeCellItems(objects: [Any]) -> [CollectionViewCellItem]
+    ///    - objects: an array of objects to create cell items for them
+    public func makeCellItems(objects: [Object]) -> [CollectionViewCellItem] {
+        objects.compactMap(makeCellItem)
+    }
 
-    /// Returns a cell items for associated object
+    /// Returns an instance of `UniversalCollectionViewCellItem` and associates provided handlers with them
     ///
     /// - Parameters:
-    ///    - object: an object associated with cell item
-    func makeCellItems(object: Any) -> [CollectionViewCellItem]
+    ///    - object: an object to create a cell item for it
+    ///    - configure: a cell configuration handler
+    ///    - size: a cell size configuration handler
+    public func makeCellItem<ConcreteCellItem: CellItem>(object: Object,
+                             configure configurationHandler: @escaping (Cell) -> Void,
+                             size sizeConfigurationHandler: @escaping (UICollectionView, CollectionViewSectionItem) -> CGSize) -> ConcreteCellItem {
+        let cellItem = ConcreteCellItem(object: object)
+        cellItem.configurationHandler = configurationHandler
+        cellItem.sizeConfigurationHandler = sizeConfigurationHandler
+        return cellItem
+    }
+
+    /// Returns an instance of `UniversalCollectionViewCellItem` and associates provided handlers with them
+    ///
+    /// - Parameters:
+    ///    - object: an object to create a cell item for it
+    public func makeCellItem<ConcreteCellItem: CellItem>(object: Object) -> ConcreteCellItem {
+        let cellItem = ConcreteCellItem(object: object)
+        basicSetup(cellItem: cellItem, object: object)
+        return cellItem
+    }
+
+    /// Assigns cell configuration and size configuration handlers to factory handlers
+    ///
+    /// - Parameters:
+    ///    - object: an object to create a cell item for it
+    public func basicSetup<ConcreteCellItem: CellItem>(cellItem: ConcreteCellItem, object: Object) {
```

Comment:
Pretty strange name for a a function? Is there other any `setup` method?

Reply:
could not remember a point of this, renamed to setup